isodatetime/parsers.py

```diff
@@ -344,7 +344,7 @@ def _parse_from_custom_regex(self, regex, data_string, dump_format=None,
         if not result:
             raise StrptimeConversionError(source, data_string)
         info = result.groupdict()
-        for property_, value in info.items():
+        for property_, value in info.copy().items():
```

Comment:
Probably my fault for introducing an inconsistent style here. We should either use `dict.copy().items()` or `list(dict.items())`.

Reply:
I think `list(dict.items())` is safer to avoid problems in case we implement `__copy__`/`__deepcopy__`/... At least I think `list(dict.items())` is less likely to have any side-effects. Fixed!